common/src/main/java/com/microsoft/hydralab/performance/parsers/AndroidBatteryInfoResultParser.java

```diff
@@ -2,17 +2,130 @@
 // Licensed under the MIT License.
 package com.microsoft.hydralab.performance.parsers;
 
+import com.microsoft.hydralab.performance.Entity.AndroidBatteryInfo;
 import com.microsoft.hydralab.performance.PerformanceInspectionResult;
 import com.microsoft.hydralab.performance.PerformanceResultParser;
 import com.microsoft.hydralab.performance.PerformanceTestResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.regex.Pattern;
 
 public class AndroidBatteryInfoResultParser implements PerformanceResultParser {
+    protected Logger classLogger = LoggerFactory.getLogger(getClass());
 
     @Override
-    public PerformanceTestResult parse(List<PerformanceInspectionResult> performanceInspectionResultList) {
-        return null;
+    public PerformanceTestResult parse(PerformanceTestResult performanceTestResult) {
+        if (performanceTestResult == null || performanceTestResult.performanceInspectionResults == null
+                || performanceTestResult.performanceInspectionResults.isEmpty()) {
+            return null;
+        }
+
+        List<PerformanceInspectionResult> inspectionResults = performanceTestResult.performanceInspectionResults;
+        for (PerformanceInspectionResult inspectionResult : inspectionResults) {
+            File logFile = inspectionResult.rawResultFile;
+            AndroidBatteryInfo batteryInfo = parseRawResultFile(logFile, inspectionResult.inspection.appId);
+            if (batteryInfo != null) {
+                batteryInfo.setAppPackageName(inspectionResult.inspection.appId);
+                batteryInfo.setDescription(inspectionResult.inspection.description);
+                batteryInfo.setTimeStamp(inspectionResult.timestamp);
+            }
+            inspectionResult.parsedData = batteryInfo;
+        }
+
+        // Use the battery usage at the end of the test as a summary
+        performanceTestResult.resultSummary = inspectionResults.get(inspectionResults.size() - 1).parsedData;
```

Comment:
I'm worried about the shallow copy (or reference) issue here, since we use the input param `performanceTestResult` to parse, but at the end we update the exact same object and return it. What if someone updates the `performanceTestResult.resultSummary` outside of this method after they call this method?

Reply:
Although Java always passes param as `value`, this doesn't mean that the reference field (non-primitive type) in param is passed by `value`, they can be changed. And it usually causes subtle bugs.

Reply:
Use deep clone instead